app/javascript/components/EmailEditor/EmailEditor.js

```diff
@@ -18,6 +18,10 @@ export default class EmailEditor extends Component {
     super(props);
     const fn = this.props.templateInterpolate || interpolateVars;
 
+    // Check whether the header/footer is configured
+    this.hasHeaderValue = this.props.header && this.hasValue(this.props.header);
+    this.hasFooterValue = this.props.footer && this.hasValue(this.props.footer);
```

Comment:
I kept it out of state since the header and footer aren't going to change based on the editor changes. If we were to keep it in state variable then it would be checking for every state change which is a definitely a repetitive operation

Reply:
Won't change from editor changes, but these are after all computed values that need to change whenever props change. I'd recommend to use a [memoization helper](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization) as recommended by the [`static getDerivedStateFromProps()` documentation](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops):

> If you want to **re-compute some data only when a prop changes**, use a [memoization helper instead](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization).

Reply:
Have made use of the memoize package